lib/bolt/transport/base.rb

```diff
@@ -164,8 +164,8 @@ def batch_connected?(targets)
       #
       # Transports may override this method, and the corresponding batch_*
       # methods, to implement their own batch processing.
-      def batches(targets)
-        targets.map { |target| [target] }
+      def batches(target_mapping)
+        target_mapping.keys.map { |target| [target] }
```

Comment:
I think we want to pass the idea of target mapping all the way down to the transport's `batch_*` methods. So this method should return a list of single-element hashes which the executor will pass directly into `batch_*`. Then `batch_*` can pull them apart and call like `run_command(target, command, ...)`.